slf4j/src/main/scala-3/org/typelevel/log4cats/slf4j/Slf4jLoggerFactory.scala

```diff
@@ -0,0 +1,48 @@
+package org.typelevel.log4cats.slf4j
+
+import cats.effect.Sync
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.internal.*
+import org.slf4j.Logger as JLogger
+import org.typelevel.log4cats.slf4j.Slf4jLogger.{
+  getLoggerFromClass,
+  getLoggerFromName,
+  getLoggerFromSlf4j
+}
+
+trait Slf4jLoggerFactory[F[_]] {
```

Comment:
Is it possible to avoid duplication? This factory trait looks the same for scala 2 and scala 3.